cdap-common/src/main/java/co/cask/cdap/common/namespace/AbstractNamespaceClient.java

```diff
@@ -37,14 +37,14 @@
   private static final Gson GSON = new Gson();
 
   @Override
-  public void delete(Id.Namespace namespaceId) throws Exception {
-    URL url = resolve(String.format("unrecoverable/namespaces/%s", namespaceId.getId()));
+  public void delete(NamespaceId namespaceId) throws Exception {
```

Comment:
One note is that, it is a client library, not part of core CDAP api. This means the user of this library doesn't have to upgrade the library version as long as the REST API that it speaks are still compatible. In longer term, I think we may better detach the client library from the CDAP core, since client library ties to CDAP REST API version, which is different than the CDAP core version.